bool brake0
bool brake1
bool fieldset0
bool fieldset1